compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -964,3 +975,39 @@ impl<'a, 'tcx> MutVisitor<'tcx> for Integrator<'a, 'tcx> {
         }
     }
 }
+
+pub(crate) fn mir_inliner_crate_fn_unexported_callees_count_map<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    _: (),
+) -> Lrc<FxHashMap<LocalDefId, usize>> {
+    let mut callers_count_map = FxHashMap::default();
+    let access_levels = tcx.privacy_access_levels(());
+
+    for callee_def_id in tcx
+        // Iterate over all functions and closures of current crate
+        .mir_fn_and_closures_keys(())
+        .iter()
+        // Get all calless in crate functions
+        .map(|def_id| {
+            tcx.mir_inliner_callees(ty::InstanceDef::Item(ty::WithOptConstParam::unknown(
+                def_id.to_def_id(),
+            )))
+        })
+        .flatten()
```

Comment:
Please prefer nested loops.

Reply:
Do we have style guide for situations like this? I wasn't able to find any.

Reply:
My personal rule is to avoid mixing functional and imperative style. This loop is imperative because it mutates `callers_count_map`, so I find an explicit loop and branches much easier to reason with.

Reply:
I changed it to full functional style 😄